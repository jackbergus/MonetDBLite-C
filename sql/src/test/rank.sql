select ROW_NUMBER() over () as foo from tables;
select ROW_NUMBER() over (PARTITION BY schema_id) as foo, schema_id from tables;
select ROW_NUMBER() over (PARTITION BY schema_id ORDER BY schema_id) as foo, schema_id from tables;
select ROW_NUMBER() over (ORDER BY schema_id) as foo, schema_id from tables;

select RANK() over () as foo from tables;
select RANK() over (PARTITION BY schema_id) as foo, schema_id from tables;
select RANK() over (PARTITION BY schema_id ORDER BY schema_id) as foo, schema_id from tables;
select RANK() over (ORDER BY schema_id) as foo, schema_id from tables;

select RANK() over () as foo, name, "type" from columns;
select RANK() over (PARTITION BY name) as foo, name, "type" from columns;
select RANK() over (PARTITION BY name ORDER BY name, "type") as foo, name, "type" from columns;
select RANK() over (ORDER BY name, "type") as foo, name, "type" from columns;

select DENSE_RANK() over () as foo, name, "type" from columns;
select DENSE_RANK() over (PARTITION BY name) as foo, name, "type" from columns;
select DENSE_RANK() over (PARTITION BY name ORDER BY name, "type") as foo, name, "type" from columns;
select DENSE_RANK() over (ORDER BY name, "type") as foo, name, "type" from columns;
