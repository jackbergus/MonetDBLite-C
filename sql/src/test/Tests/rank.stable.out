stdout of test 'rank` in directory 'src/test` itself:


# 19:18:12 >  
# 19:18:12 >  Mtimeout -timeout 180 Mserver "--config=/ufs/niels/scratch/rc/Linux-x86_64/etc/MonetDB.conf" --debug=10 --set "monet_mod_path=/ufs/niels/scratch/rc/Linux-x86_64/lib/MonetDB4:/ufs/niels/scratch/rc/Linux-x86_64/lib/MonetDB4/lib:/ufs/niels/scratch/rc/Linux-x86_64/lib/MonetDB4/bin" --set "gdk_dbfarm=/ufs/niels/scratch/rc/Linux-x86_64/var/MonetDB4/dbfarm" --set "sql_logdir=/ufs/niels/scratch/rc/Linux-x86_64/var/MonetDB4/sql_logs" --set "xquery_logdir=/ufs/niels/scratch/rc/Linux-x86_64/var/MonetDB4/xquery_logs" --set mapi_port=33222 --set xrpc_port=48072 --set monet_prompt= --trace "--dbname=mTests_src_test" --dbinit=" module(sql_server);" ; echo ; echo Over..
# 19:18:12 >  

# Monet Database Server V4.16.3
# Copyright (c) 1993-2007, CWI. All rights reserved.
# Compiled for x86_64-redhat-linux-gnu/64bit with 64bit OIDs; dynamically linked.
# Visit http://monetdb.cwi.nl/ for further information.

printf("\nReady.\n");

Ready.
quit();

Over..

# 19:18:12 >  
# 19:18:12 >  Mtimeout -timeout 60 MapiClient -lsql -umonetdb -Pmonetdb --host=localhost --port=33222 
# 19:18:12 >  

% sys. # table_name
% foo # name
% int # type
% 2 # length
[ 0	]
[ 1	]
[ 2	]
[ 3	]
[ 4	]
[ 5	]
[ 6	]
[ 7	]
[ 8	]
[ 9	]
[ 10	]
[ 11	]
[ 12	]
[ 13	]
[ 14	]
[ 15	]
[ 16	]
[ 17	]
[ 18	]
[ 19	]
[ 20	]
[ 21	]
[ 22	]
[ 23	]
[ 24	]
[ 25	]
[ 26	]
[ 27	]
[ 28	]
[ 29	]
[ 30	]
[ 31	]
% sys.tables,	sys.tables # table_name
% foo,	schema_id # name
% int,	int # type
% 2,	4 # length
[ 0,	956	]
[ 1,	956	]
[ 2,	956	]
[ 3,	956	]
[ 4,	956	]
[ 5,	956	]
[ 6,	956	]
[ 7,	956	]
[ 8,	956	]
[ 9,	956	]
[ 10,	956	]
[ 11,	956	]
[ 12,	956	]
[ 0,	1057	]
[ 1,	1057	]
[ 2,	1057	]
[ 3,	1057	]
[ 4,	1057	]
[ 5,	1057	]
[ 13,	956	]
[ 14,	956	]
[ 15,	956	]
[ 16,	956	]
[ 17,	956	]
[ 18,	956	]
[ 19,	956	]
[ 6,	1057	]
[ 20,	956	]
[ 21,	956	]
[ 22,	956	]
[ 23,	956	]
[ 24,	956	]
% sys.tables,	sys.tables # table_name
% foo,	schema_id # name
% int,	int # type
% 2,	4 # length
[ 0,	956	]
[ 1,	956	]
[ 2,	956	]
[ 3,	956	]
[ 4,	956	]
[ 5,	956	]
[ 6,	956	]
[ 7,	956	]
[ 8,	956	]
[ 9,	956	]
[ 10,	956	]
[ 11,	956	]
[ 12,	956	]
[ 0,	1057	]
[ 1,	1057	]
[ 2,	1057	]
[ 3,	1057	]
[ 4,	1057	]
[ 5,	1057	]
[ 13,	956	]
[ 14,	956	]
[ 15,	956	]
[ 16,	956	]
[ 17,	956	]
[ 18,	956	]
[ 19,	956	]
[ 6,	1057	]
[ 20,	956	]
[ 21,	956	]
[ 22,	956	]
[ 23,	956	]
[ 24,	956	]
% .,	sys.tables # table_name
% foo,	schema_id # name
% int,	int # type
% 2,	4 # length
[ 0,	956	]
[ 1,	956	]
[ 2,	956	]
[ 3,	956	]
[ 4,	956	]
[ 5,	956	]
[ 6,	956	]
[ 7,	956	]
[ 8,	956	]
[ 9,	956	]
[ 10,	956	]
[ 11,	956	]
[ 12,	956	]
[ 13,	956	]
[ 14,	956	]
[ 15,	956	]
[ 16,	956	]
[ 17,	956	]
[ 18,	956	]
[ 19,	956	]
[ 20,	956	]
[ 21,	956	]
[ 22,	956	]
[ 23,	956	]
[ 24,	956	]
[ 25,	1057	]
[ 26,	1057	]
[ 27,	1057	]
[ 28,	1057	]
[ 29,	1057	]
[ 30,	1057	]
[ 31,	1057	]
% sys. # table_name
% foo # name
% int # type
% 2 # length
[ 0	]
[ 1	]
[ 2	]
[ 3	]
[ 4	]
[ 5	]
[ 6	]
[ 7	]
[ 8	]
[ 9	]
[ 10	]
[ 11	]
[ 12	]
[ 13	]
[ 14	]
[ 15	]
[ 16	]
[ 17	]
[ 18	]
[ 19	]
[ 20	]
[ 21	]
[ 22	]
[ 23	]
[ 24	]
[ 25	]
[ 26	]
[ 27	]
[ 28	]
[ 29	]
[ 30	]
[ 31	]
% sys.tables,	sys.tables # table_name
% foo,	schema_id # name
% int,	int # type
% 2,	4 # length
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 13,	1057	]
[ 13,	1057	]
[ 13,	1057	]
[ 13,	1057	]
[ 13,	1057	]
[ 13,	1057	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 13,	1057	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
% sys.tables,	sys.tables # table_name
% foo,	schema_id # name
% int,	int # type
% 2,	4 # length
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 13,	1057	]
[ 13,	1057	]
[ 13,	1057	]
[ 13,	1057	]
[ 13,	1057	]
[ 13,	1057	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 13,	1057	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
% .,	sys.tables # table_name
% foo,	schema_id # name
% int,	int # type
% 2,	4 # length
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 0,	956	]
[ 16,	1057	]
[ 16,	1057	]
[ 16,	1057	]
[ 16,	1057	]
[ 16,	1057	]
[ 16,	1057	]
[ 16,	1057	]

# 22:17:03 >  
# 22:17:03 >  Done.
# 22:17:03 >  

